File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs (computers, start, visited):
    visited.append(start)
    for i in range(len(computers[start])):
        if i != start and i not in visited and computers[start][i] == 1:
            dfs(computers, i, visited)
    
def solution(n, computers):
    answer = 0
    visited = []
    for i in range(n):
        if i not in visited:
            dfs(computers, i, visited)
            answer+=1
            
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited = []
    for(let i=0; i < n ; i++){
        if (!visited.includes(i)){
            dfs(computers, i , visited)
            answer ++
        }
    }
    return answer;
}

function dfs(computers, start , visited) {
    visited.push(start);
    computers[start].map((v,i)=>{
        if(i !== start && !visited.includes(i) && v === 1){
            dfs(computers, i, visited)
        }
    })
}